packages/block-editor/src/components/global-styles/hooks.js

```diff
@@ -75,22 +135,51 @@ export function useGlobalSetting( path, blockName, source = 'all' ) {
 	};
 
 	// Unlike styles settings get inherited from top level settings.
-	const resultWithFallback =
-		getSettingValueForContext( blockName ) ?? getSettingValueForContext();
+	const resultWithFallback = useMemo( () => {
+		if ( propertyPath ) {
+			return (
+				getSettingValueForContext( blockName, propertyPath ) ??
+				getSettingValueForContext( undefined, propertyPath )
+			);
+		}
+		const result = {};
+		VALID_SETTINGS.forEach( ( setting ) => {
+			set(
+				result,
+				setting,
+				getSettingValueForContext( blockName, setting ) ??
+					getSettingValueForContext( undefined, setting )
+			);
+		} );
+		return result;
+	}, [
+		mergedConfig,
+		userConfig,
+		baseConfig,
+		blockName,
+		propertyPath,
+		source,
+	] );
 
 	return [ resultWithFallback, setSetting ];
 }
 
-export function useGlobalStyle( path, blockName, source = 'all' ) {
+export function useGlobalStyle(
+	path,
```

Comment:
Not really because it's not just a propertyPath (like settings) but it can also contain a prefix (variation or element). I think the ideal API would split these into different arguments and the concatenation should happen with the function though.

Reply:
I dislike the inconsistency between `useGlobalStyle` and `useGlobalSetting` but okay.